SPECS/kernel-headers/kernel-headers.spec

```diff
@@ -1,41 +1,80 @@
+# This package doesn't contain any binaries, thus no debuginfo package is needed.
+%global debug_package %{nil}
+
+%if "%{_arch}" == "x86_64"
+%define cross_archs arm64
+%else
+%define cross_archs %{nil}
+%endif
+
 Summary:        Linux API header files
 Name:           kernel-headers
 Version:        5.15.138.1
-Release:        1%{?dist}
+Release:        2%{?dist}
 License:        GPLv2
 Vendor:         Microsoft Corporation
 Distribution:   Mariner
 Group:          System Environment/Kernel
 URL:            https://github.com/microsoft/CBL-Mariner-Linux-Kernel
-#Source0:       https://github.com/microsoft/CBL-Mariner-Linux-Kernel/archive/rolling-lts/mariner-2/%%{version}.tar.gz
-Source0:        kernel-%{version}.tar.gz
+Source0:        https://github.com/microsoft/CBL-Mariner-Linux-Kernel/archive/rolling-lts/mariner-2/%{version}.tar.gz#/kernel-%{version}.tar.gz
 # Historical name shipped by other distros
 Provides:       glibc-kernheaders = %{version}-%{release}
 BuildArch:      noarch
 
 %description
 The Linux API Headers expose the kernel's API for use by Glibc.
 
+%if "%{_arch}" == "x86_64"
+%package -n kernel-cross-headers
```

Comment:
Will the cross package always get built and will we want to make that consistent with all other cross packages? It could add a significant amount of build time if so, especially for glibc and gcc.

And does it make sense to keep this package definition consistent with the cross-binutils design of moving it near the top of the file with macros?

Reply:
This package's build time is pretty fast, but glibc and gcc are quite long. I'm wondering if it's possible for the cross packages to be conditionally built (besides relying on the host arch). From what I understand here it looks like both the native and cross version of kernel-headers get built for AMD64. Will glibc and gcc follow this pattern and also build the native and cross packages? This isn't meant to be a blocking comment, but might be a question for the next iteration.

Reply:
We don't have any mechanism right now to conditionally disabled building subpackages within one architecture, so yes, we will always build the cross-compiling subpackages on AMD64. But that is also the way all other package builds work. The additional overhead here or in other packages is currently not a big worry.

Having said that, could you say scenario are you worried about in this case? I might be missing a crucial concern.

Reply:
As long as the additional build time isn't an issue this is a good approach.